### ISSUE TITLE
Connection refused for (at least) B2 and AWS S3 backends since 0.13.0

### ISSUE THREAD

**Issue by aaronkyriesenbach** [restic/restic#3777]
Output of `restic version`
--------------------------
`restic 0.13.1 compiled with go1.18.1 on linux/amd64`

How did you run restic exactly?
-------------------------------
Command:
`restic snapshots`

Environment:
```
AWS_ACCESS_KEY_ID={id}
AWS_SECRET_ACCESS_KEY={key}
AWS_DEFAULT_REGION="us-east-1"
RESTIC_REPOSITORY={repository}
RESTIC_PASSWORD={password}
```

Output:
```
Fatal: unable to open config file: Stat: Head "https://{repository}.s3.dualstack.us-east-1.amazonaws.com/config": dial tcp: lookup {repository}.s3.dualstack.us-east-1.amazonaws.com on [::1]:53: read udp [::1]:38451->[::1]:53: read: connection refused
Is there a repository at the following location?
s3:s3.amazonaws.com/{repository}
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
AWS S3

Expected behavior
-----------------
Restic successfully opens repository and lists snapshots as it does using version 0.12.1 on the same system:
`repository d993ac91 opened successfully, password is correct`

Actual behavior
---------------
Restic fails to open the repository and gives a connection refused error

Steps to reproduce the behavior
-------------------------------
1. Install restic version >= 0.13.0
2. Attempt to do anything using a remote S3 or B2 repository. This issue was encountered while attempting `restic snapshots` and `restic init`, but I assume that it occurs for other commands as well.

Do you have any idea what may have caused this?
-----------------------------------------------
No

Do you have an idea how to solve the issue?
-------------------------------------------
No, sorry (other than downgrading to 0.12.1)

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
Yes! I'm excited to be working on backing up my files again. (Stopped in February while transitioning between systems and starting back up again now.) I very much enjoy using Restic and it's been great for all of my use up till now. Thanks devs!

**Comment by MichaelEischer** [comment]
This looks like a DNS (port 53) problem on your host:
```
dial tcp: lookup {repository}.s3.dualstack.us-east-1.amazonaws.com on [::1]:53: read udp [::1]:38451->[::1]:53: read: connection refused
```

Apparently your new system is configured to have a DNS server running on localhost at port 53, but there is none or restic is unable to connect.

**Comment by aaronkyriesenbach** [comment]
@MichaelEischer While that could make sense, why would the issue occur in 0.13.0 and onwards but not in earlier versions? I’ll be able to share my DNS config when I’m home later today.

**Comment by MichaelEischer** [comment]
My guess would be that something has changed in the go DNS resolver. restic 0.13.1 uses go 1.18 whereas restic 0.12.1 was originally built using go1.16.

> I do use systemd-resolved which uses port 53.

resolved normally uses 127.0.0.53 as nameserver. However, the above error message complained about `::1`. Apparently, go falls back to 127.0.0.1 and ::1 if the resolv.conf cannot be read.

**Comment by aaronkyriesenbach** [comment]
Got it. Looks like the resolver now uses `/etc/resolv.conf` instead of whatever it was doing before, so I had to configure the stub mode symlink and the issue is now fixed. Thank you for the info!
